dev_tools/container/Dockerfile

```diff
@@ -0,0 +1,98 @@
+FROM ubuntu:22.04
+
+# install spack prerequisites and basics
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+    build-essential \
+    ca-certificates \
+    coreutils \
+    curl \
+    environment-modules \
+    gfortran \
+    git \
+    gpg \
+    gpg-agent \
+    lsb-release \
+    python3 \
+    python3-distutils \
+    python3-venv \
+    unzip \
+    zip \
+    vim-nox \
+    wget \
+    libopencv-dev \
+    pkg-config \
+    ninja-build \
+    # Dependencies for our spack packages
+    autoconf automake bzip2 diffutils libtool m4 tar xz-utils perl pigz re2c openssh-client \    
+    && rm -rf /var/lib/apt/lists/*
+
+RUN (echo "deb http://apt.llvm.org/jammy/ llvm-toolchain-jammy-15 main" > /etc/apt/sources.list.d/llvm15.list) && \
+    (curl https://apt.llvm.org/llvm-snapshot.gpg.key > /etc/apt/trusted.gpg.d/apt.llvm.org.asc) && apt-get update && \
+    apt-get install -y --no-install-recommends clang-15 clang-tools-15 clang-15-doc clangd-15 clang-tidy-15 clang-format-15 \
```

Comment:
should it use format 14 because we use clang-format-14 as our standard in repo

Reply:
I think we could install just clang-format-14 and use 15 for the rest, I would like to profit from the latest clang features